pkg/catalogv2/roundtripper/roundtripper.go

```diff
@@ -0,0 +1,91 @@
+package roundtripper
+
+import (
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+type DisableSameOriginRoundTripper struct {
```

Comment:
The struct and the New... `func` requires documentation that explicitly calls out the dangers of using this and why it should be avoided in production settings.

```suggestion
type DisableSameOriginRoundTripper struct {
```

Reply:
Done.